nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EncodeContent.java

```diff
@@ -129,31 +175,35 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         }
     }
 
-    private static StreamCallback getStreamCallback(final boolean encode, final String encoding) {
-        if (encode) {
-            if (encoding.equalsIgnoreCase(BASE64_ENCODING)) {
-                return new EncodeBase64();
-            } else if (encoding.equalsIgnoreCase(BASE32_ENCODING)) {
-                return new EncodeBase32();
-            } else {
-                return new EncodeHex();
-            }
-        } else {
-            if (encoding.equalsIgnoreCase(BASE64_ENCODING)) {
-                return new DecodeBase64();
-            } else if (encoding.equalsIgnoreCase(BASE32_ENCODING)) {
-                return new DecodeBase32();
-            } else {
-                return new DecodeHex();
-            }
+    private static StreamCallback getStreamCallback(final Boolean encode, final String encoding,
+        final int lineLength, final String lineSeparator) {
+        switch(encoding) {
+            case "BASE64_ENCODING":
+                return encode ? new EncodeBase64(lineLength, lineSeparator) : new DecodeBase64();
+            case "BASE32_ENCODING":
+                return encode ? new EncodeBase32(lineLength, lineSeparator) : new DecodeBase32();
```

Comment:
```suggestion
    private static StreamCallback getStreamCallback(final Boolean encode, final EncodingType encoding,
        final int lineLength, final String lineSeparator) {
        switch(encoding) {
            case BASE64_ENCODING:
                return encode ? new EncodeBase64(lineLength, lineSeparator) : new DecodeBase64();
            case BASE32_ENCODING:
                return encode ? new EncodeBase32(lineLength, lineSeparator) : new DecodeBase32();
```

Reply:
Interesting.  Okay, didn't know enums behaved that way.  Committing.